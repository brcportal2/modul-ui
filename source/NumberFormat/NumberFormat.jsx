import React, {Component} from 'react';
import PropTypes from 'prop-types';
import accounting from 'accounting';
import {
    validateHelper,
    numberHelper
} from 'modul-helpers';


accounting.settings = {
    number: {
        decimal: ','
    }
};

export default class NumberFormat extends Component {
    static propTypes = {
        value: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number
        ]),
        def: PropTypes.string,
        className: PropTypes.string
    };

    static defaultProps = {
        def: '',
        className: ''
    };

    getPrecision = value => {
        const valueStr = this.clean(value);
        const pos = valueStr.indexOf(',');

        return pos > -1 ? valueStr.length - pos - 1 : 0;
    };

    clean = val => {
        let value = val;

        if (!value.replace)
            value = value.toString();

        return numberHelper.cleanValue(value);
    };

    render() {
        const {
            value,
            def,
            ...props
        } = this.props;

        if (validateHelper.isEmpty(value))
            return def ? (
                <span
                    {...props}
                >{def}</span>
            ) : null;

        const precision = this.getPrecision(value);
        const formatted = accounting.formatNumber(value, precision, ' ');

        return (
            <span
                {...props}
            >{formatted}</span>
        );
    }
}