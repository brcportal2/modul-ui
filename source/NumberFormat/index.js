import NumberFormat from './NumberFormat';


export default NumberFormat;
export {
	NumberFormat
};