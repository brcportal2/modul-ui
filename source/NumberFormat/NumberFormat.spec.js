import React from 'react';
import renderer from 'react-test-renderer';

import NumberFormat from './NumberFormat';


describe('NumberFormat', () => {
    test('Default props', () => {
        const component = renderer.create(
            <NumberFormat
                value='123456'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Default value', () => {
        const component = renderer.create(
            <NumberFormat
                def='not date'
                value='123456'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('ClassName', () => {
        const component = renderer.create(
            <NumberFormat
                className='test_class'
                def='not date'
                value='123456'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Big value', () => {
        const component = renderer.create(
            <NumberFormat
                className='test_class'
                def='not date'
                value='1234567890'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('First zero', () => {
        const component = renderer.create(
            <NumberFormat
                className='test_class'
                def='not date'
                value='0123'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Null value', () => {
        const component = renderer.create(
            <NumberFormat
                className='test_class'
                def='not date'
                value={null}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});