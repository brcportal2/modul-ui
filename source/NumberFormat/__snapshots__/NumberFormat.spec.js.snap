// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberFormat Big value 1`] = `
<span
  className="test_class"
>
  1 234 567 890
</span>
`;

exports[`NumberFormat ClassName 1`] = `
<span
  className="test_class"
>
  123 456
</span>
`;

exports[`NumberFormat Default props 1`] = `
<span
  className=""
>
  123 456
</span>
`;

exports[`NumberFormat Default value 1`] = `
<span
  className=""
>
  123 456
</span>
`;

exports[`NumberFormat First zero 1`] = `
<span
  className="test_class"
>
  123
</span>
`;

exports[`NumberFormat Null value 1`] = `
<span
  className="test_class"
>
  not date
</span>
`;
